From 919786768a42cf66a538d831dbecc927caa8cb1c Mon Sep 17 00:00:00 2001
From: Venkat Chimata <venkat@nearhop.com>
Date: Wed, 11 Feb 2026 20:11:15 +0530
Subject: [PATCH] hostapd/wifi-7: fix RRM neighbor report value not updating on
 hot reload

 - Ensure rrm_neighbor_report is always set based on the current configuration during hot reload.
 - Prevent previously enabled values from being unintentionally preserved.
 - Allows both enable and disable transitions to take effect correctly without requiring a full restart.

Signed-off-by: Venkat Chimata <venkat@nearhop.com>
---
 package/network/services/hostapd/files/hostapd.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/package/network/services/hostapd/files/hostapd.sh b/package/network/services/hostapd/files/hostapd.sh
index 958d5a9ac6..2abfe215a9 100644
--- a/package/network/services/hostapd/files/hostapd.sh
+++ b/package/network/services/hostapd/files/hostapd.sh
@@ -905,8 +905,8 @@ hostapd_set_bss_options() {
 		set_default rrm_beacon_report 0
 	fi
 
-	[ "$rrm_neighbor_report" -eq "1" ] && append bss_conf "rrm_neighbor_report=1" "$N"
-	[ "$rrm_beacon_report" -eq "1" ] && append bss_conf "rrm_beacon_report=1" "$N"
+	append bss_conf "rrm_neighbor_report=$rrm_neighbor_report" "$N"
+	append bss_conf "rrm_beacon_report=$rrm_beacon_report" "$N"
 
 	json_get_vars ftm_responder stationary_ap lci civic
 	set_default ftm_responder 0
-- 
2.34.1

